invoices/invoices.go

```diff
@@ -707,10 +725,7 @@ func ValidateInvoice(i *Invoice, paymentHash lntypes.Hash) error {
 
 	// AMP invoices and hodl invoices are allowed to have no preimage
 	// specified.
-	isAMP := i.Terms.Features.HasFeature(
-		lnwire.AMPOptional,
-	)
-	if i.Terms.PaymentPreimage == nil && !(i.HodlInvoice || isAMP) {
+	if i.Terms.PaymentPreimage == nil && !(i.HodlInvoice || i.IsAMP()) {
```

Comment:
While we're here it could be nice to add a `RequirePreimage` helper that covers the `!(i.HodlInvoice || i.IsAMP())` condition?